src/dlsnode/storage/iterator/NeoStorageEngineStepIterator.d

```diff
@@ -342,20 +350,34 @@ public class NeoStorageEngineStepIterator
                         // move to the next bucket
                         this.file.close();
                         this.state = State.LookingForNextBucket;
-                        break;
                     }
                     else
                     {
-                        key = cast(time_t)this.record_header.key;
-                        value.copy(this.value_future.get());
-                        this.state = State.ExpectingRecordHeader;
-                        return NextResult.RecordRead;
+                        try
+                        {
+                            key = cast(time_t)this.record_header.key;
+                            value.copy(this.value_future.get());
+                            this.state = State.ExpectingRecordHeader;
+                            return NextResult.RecordRead;
+                        }
+                        catch (Exception e)
+                        {
+                            // on error reading, skip this bucket
+                            this.state = State.LookingForNextBucket;
+                        }
+                        break;
                     }
```

Comment:
I think `break;` should go after this curly bracket instead

Reply:
Of course it should, thanks!